src/components/DataTable/cells/TableCell.tsx

```diff
@@ -1,32 +1,90 @@
 import React from 'react';
 import clsx from 'clsx';
+import { IconTextIcon, Typography } from '~/components';
+import { ChevronDown, ChevronRight } from '~/icons';
 
-import type {TableCellProps} from './TableCell.types';
-import {capitalize} from '~/utils/helpers';
+import type { TableCellProps } from './TableCell.types';
+import { capitalize } from '~/utils/helpers';
 import './TableCell.scss';
 
 /**
- * Base table cell component.
- *
- * Usage 1: Direct usage with children prop → renders children directly.
+ * Enhanced TableCell component that handles structured view, icons, and typography.
+ * Merges functionality from the TableBodyCell.
  */
 const TableCellForwardRef: React.ForwardRefRenderFunction<HTMLTableCellElement, TableCellProps> = (
-    {className, children, textAlign, ...props},
+    {
+        className,
+        children,
+        textAlign = 'left',
+        verticalAlign = 'center',
+        iconStart,
+        iconEnd,
+        isExpandableColumn,
+        row,
+        width,
+        isScrollable,
+        component = 'td',
+        ...props
+    },
     ref
 ) => {
+    const leftMarginBuffer = 20; // Px
+    const leftMarginIndentDepth = (row?.depth ?? 0) * 20; // Px
+    const scrollableClass = isScrollable ? 'moonstone-tableCellContent' : '';
+
+    const renderCellContent = () => (
+        <IconTextIcon component="div" iconStart={iconStart} iconEnd={iconEnd} typographyProps={{ className: clsx(scrollableClass, 'flexFluid') }}>
+            {children ?? '-'}
+        </IconTextIcon>
+    );
+
+    const renderInnerContent = () => {
```

Comment:
Maybe it makes more sense to create `renderExpandableContent` to manage `option 1` and `option 2` since option 3 is very straightforward.